lib/registry/src/lib.rs

```diff
@@ -347,6 +350,15 @@ pub enum Registries {
     Multi(MultiRegistry),
 }
 
+impl Registries {
+    pub fn current(&self) -> &str {
```

Comment:
I've copied the `impl Registries` block across. You'll probably get some small merge conflicts because I replaced `println!()` statements with `log::warn!()` and friends - libraries shouldn't be printing to stdout.